src/test/java/org/icgc/argo/program_service/services/UpdateProgramTest.java

```diff
@@ -0,0 +1,136 @@
+
+  @Test
+  public void add_cancer_primary_site_success(){
+    val programToUpdate = entityGenerator.setUpProgramEntity("PROGRAM-CA");
+    val programId = programToUpdate.getId();
+    val updatingCancer = entityGenerator.setUpCancer("Soft Tissue cancer");
+    val updatingPrimarySite = entityGenerator.setUpPrimarySite("Liver");
+
+    val existingCancers = cancerRepository.findAll(CancerSpecification.containsProgram(programId));
+    val existingPrimarySites = primarySiteRepository.findAll(PrimarySiteSpecification.containsProgram(programId));
+
+    assertThat(existingCancers.isEmpty()).isTrue();
+    assertThat(existingPrimarySites.isEmpty()).isTrue();
+
+    val cancers = List.of("Soft Tissue cancer");
+    val primarySites = List.of("Liver");
+    programService.updateProgram(programToUpdate, cancers, primarySites);
+
+    val updatedCancers = cancerRepository.findAll(CancerSpecification.containsProgram(programId));
+    val updatedPrimarySites = primarySiteRepository.findAll(PrimarySiteSpecification.containsProgram(programId));
+
+    assertThat(updatedCancers.size()).isEqualTo(1);
+    assertThat(updatedCancers.contains(updatingCancer)).isTrue();
+
+    assertThat(updatedPrimarySites.size()).isEqualTo(1);
+    assertThat(updatedPrimarySites.contains(updatingPrimarySite)).isTrue();
+  }
+
+  @Test
+  public void dissociate_all_success(){
+    val programToUpdate = entityGenerator.setUpProgramEntity("PROGRAM-CA");
+    val programId = programToUpdate.getId();
+    entityGenerator.setUpCancer("Soft Tissue cancer");
+    entityGenerator.setUpCancer("Brain cancer");
+    entityGenerator.setUpPrimarySite("Liver");
+    entityGenerator.setUpPrimarySite("Brain");
+
+    val cancers = List.of("Soft Tissue cancer", "Brain cancer");
+    val primarySites = List.of("Liver", "Brain");
+    programService.updateProgram(programToUpdate, cancers, primarySites);
+
+    val cancersBeforeUpdate = cancerRepository.findAll(CancerSpecification.containsProgram(programId));
+    val primarySiteBeforeUpdate = primarySiteRepository.findAll(PrimarySiteSpecification.containsProgram(programId));
+
+    assertThat(cancersBeforeUpdate.size()).isEqualTo(2);
+    assertThat(cancersBeforeUpdate.stream().map(CancerEntity::getName).collect(toList()).containsAll(cancers)).isTrue();
+
+    assertThat(primarySiteBeforeUpdate.size()).isEqualTo(2);
+    assertThat(primarySiteBeforeUpdate.stream().map(PrimarySiteEntity::getName).collect(toList()).containsAll(primarySites)).isTrue();
+
+    programService.updateProgram(programToUpdate, Collections.EMPTY_LIST, Collections.EMPTY_LIST);
+    val cancersAfterUpdate = cancerRepository.findAll(CancerSpecification.containsProgram(programId));
+    val primarySiteAfterUpdate = primarySiteRepository.findAll(PrimarySiteSpecification.containsProgram(programId));
+
+    assertTrue(cancersAfterUpdate.isEmpty());
+    assertTrue(primarySiteAfterUpdate.isEmpty());
+  }
+
+  @Test
+  public void remove_cancer_add_primary_site_success(){
```

Comment:
all these tests are happy path tests. Create another test to catch the case when a non-existing cancer and/or a non-existing primary site is used to update a program